### ISSUE TITLE
some accesses to p.sanbox are not protected

### ISSUE THREAD

**Issue by laijs** [hyperhq/hyperd#612]
p.cleanup()/p.decommissionResources() set p.sanbox to nil.
but some code access to p.sanbox without protection.(example: p.waitVMStop())

here was panic caused by it:
```
START: hyper_test.go:390: TestSuite.TestSetPodLabels
I1214 03:21:33.290207   23312 pod.go:15] PodCreate with request podSpec:<id:"busybox" containers:<image:"busybox" > > 
I1214 03:21:33.290307   23312 factory.go:94] configuring log driver [json-file] for busybox
I1214 03:21:33.290337   23312 sandbox.go:30] get sandbox from factory: CPU: 1, Memory 128
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x687ac6]

goroutine 825 [running]:
panic(0x1195500, 0xc4200121b0)
	/home/travis/.gimme/versions/go1.7.linux.amd64/src/runtime/panic.go:500 +0x1a1
github.com/hyperhq/runv/hypervisor.(*Vm).GetResponseChan(0x0, 0x1, 0xc42189acc0, 0xd40003)
	/home/travis/gopath/src/github.com/hyperhq/hyperd/Godeps/_workspace/src/github.com/hyperhq/runv/hypervisor/vm.go:35 +0x26
github.com/hyperhq/runv/hypervisor.(*Vm).WaitVm(0x0, 0xffffffffffffffff, 0x0)
	/home/travis/gopath/src/github.com/hyperhq/hyperd/Godeps/_workspace/src/github.com/hyperhq/runv/hypervisor/vm.go:144 +0x8e
github.com/hyperhq/hyperd/daemon/pod.(*XPod).waitVMStop(0xc42130bec0)
	/home/travis/gopath/src/github.com/hyperhq/hyperd/daemon/pod/decommission.go:541 +0x8e
created by github.com/hyperhq/hyperd/daemon/pod.(*XPod).createSandbox
	/home/travis/gopath/src/github.com/hyperhq/hyperd/daemon/pod/provision.go:241 +0x2b8
2016/12/14 03:21:33 transport: http2Client.notifyError got notified that the client transport was broken EOF.
2016/12/14 03:21:34 grpc: Conn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:22318: getsockopt: connection refused"; Reconnecting to "127.0.0.1:22318"
2016/12/14 03:21:35 grpc: Conn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:22318: getsockopt: connection refused"; Reconnecting to "127.0.0.1:22318"
2016/12/14 03:21:35 grpc: Conn.transportMonitor exits due to: grpc: timed out trying to connect
hyper_test.go:401:
    c.Assert(err, IsNil)
... value grpc.rpcError = grpc.rpcError{code:0xd, desc:"transport is closing"} ("rpc error: code = 13 desc = \"transport is closing\"")

FAIL: hyper_test.go:390: TestSuite.TestSetPodLabels

START: hyper_test.go:357: TestSuite.TestStartAndStopPod
hyper_test.go:368:
    c.Assert(err, IsNil)
... value grpc.rpcError = grpc.rpcError{code:0x2, desc:"grpc: the client connection is closing"} ("rpc error: code = 2 desc = \"grpc: the client connection is closing\"")

FAIL: hyper_test.go:357: TestSuite.TestStartAndStopPod
```

**Comment by gnawux** [comment]
We should protect it.

And I wonder what's the test context. the `p.waitVMStop()` is called quite close to where p.sandbox is assigned, and `p.decommissionResources()` is a subsequent step of `p.waitVmStop()`. 

`p.waitVmStop()` --> `p.cleanup()` --> `p.decommissionResources()`, i.e.  `p.sandbox = nil` in `p.decommissionResources()` should not cause the panic in `p.waitVmStop()`.

Is this possible: the `sb, err = startSandbox()` returns `nil, nil`, i.e. no error,  no vm either. @laijs

**Comment by gnawux** [comment]
could we close this issue @laijs ?

**Comment by gnawux** [comment]
The panic is fixed by #615. If you think we need further action on it, please file a new issue
